public/components/health-check/lib/check-time-filter.ts

```diff
@@ -0,0 +1,42 @@
+import { AxiosResponse } from 'axios';
+import { GenericRequest } from '../../../react-services';
+
+type userValue<T> = {userValue: T}
+
+type kbnSettings = {
+  buildNum: userValue<number>
+  timeFilter?: userValue<string[]>,
+};
+type responseKbnSettings = {settings: kbnSettings};
+
+const timeFilterSetting = [
+    { from: "nowtime-24h" },
+    { to: 'now'}];
+
+export function checkKibanaSettingsTimeFilter (removeMetaFields: boolean) {
+  removeMetaFields && getKibanaSettings()
+  .then(checktimeFilter)
+  .then(updateTimeFilterSetting)
+  .catch(error => error !== 'Unable to update config' && console.log(error));
+}
+
+async function getKibanaSettings(): Promise<responseKbnSettings> {
+  const kibanaSettings:AxiosResponse = await GenericRequest.request('GET', '/api/kibana/settings');
+  return kibanaSettings.data;
+}
+
+async function checktimeFilter({settings}: responseKbnSettings) {
+  const { timeFilter } = settings;
+  return !!timeFilter && !!timeFilter.userValue.length;
```

Comment:
maybe you should validate if the current value is different from to `timeFilterSetting`

Reply:
The setting is `timepicker:timeDefaults` not `timeFilter`. 

I got this response if set a different value for `timepicker:timeDefaults`, using Kibana in Stack Management > Kibana > Advanced Settings:

```
{
  settings: {
    buildNum: {
      userValue: 9007199254740991
    },
    metaFields: {
      userValue: [
        "_source",
        ""
      ]
    },
   defaultIndex: {
     userValue: "metrics-*"
   },
   timepicker:timeDefaults: {
     userValue: "{ "from": "now-24m", "to": "now" }"
   }
 }
}

```

It seems, that if the value of this setting is not the same string that default, it will not be included in the response of `/api/kibana/settings` response.